versions: update kernel to 6.16.1

arvindskumar99 opened the pull request:
Updating stable-commits to provide a way for
users to install the latest supported kernel
(v6.16.1) to match what is used in Kata and
CoCo.

ryansavino committed:
Fix threading in edk2 build of ovmf for bergamo systems

A race condition exists in edk2 build - set threads to 1 for now
Previous value: -n $(getconf _NPROCESSORS_ONLN)
Only seen the error on Bergamo systems

Signed-Off-By: Ryan Savino <ryan.savino@amd.com>

ryansavino committed:
ovmf: use the AmdSev package for kernel-hashes

switch to AmdSev package for kernel-hashes.
only build the one OVMF.fd instead of the CODE and VARS files separated.

Signed-Off-By: Ryan Savino <ryan.savino@amd.com>

ryansavino committed:
kernel build speed optimizations

Optimized kernel cloning and git commands to reduce build time.

Signed-Off-By: Ryan Savino <ryan.savino@amd.com>

ryansavino committed:
accommodate setting tag in branch variable in stable-commits

Signed-Off-By: Ryan Savino <ryan.savino@amd.com>

LakshmiSaiHarika committed:
install.sh: Fix for the virt group in the non-debian OS

virt group package is not present in the RHEL,so replacing this group package with the RHEL's qemu-kvm package

Signed-off-by: Harika <lnittala@amd.com>

LakshmiSaiHarika committed:
build.sh: Fix to copy host and guest kernel rpm packages

Copy the host kernel package and guest kernel rpm packages into the separate folders for the non-debian OS.

ryansavino committed:
versions: set stable commit versions to upstream repos and tags

Point stable-commits to the repos and tags from upstream repositories.

Signed-Off-By: Ryan Savino <ryan.savino@amd.com>

ryansavino committed:
Merge pull request #1 from LakshmiSaiHarika/rhel-setup-host

RHEL dependencies to setup latest SNP kernel on the host

LakshmiSaiHarika committed:
ovmf-fix: Workaround Zeex repo becoming private

Added ovmf fix for Zeex repo clone issue from Fabiano kata-container commit(aff3d98) during ovmf build

Signed-off-by: Harika Nittala <lnittala@amd.com>

ryansavino committed:
Merge pull request #2 from LakshmiSaiHarika/amd-snp-ovmf-fix

Added ovmf fix  for Zeex repo clone issue

mdroth committed:
launch-qemu: remove -certs options

Certificate support has been pulled out of current SNP host kernel. This
will get re-enabled later with newer code.

Signed-off-by: Michael Roth <michael.roth@amd.com>

mdroth committed:
allow using environment variables for specifying template kernel config

By default the scripts inherit from the currently running kernel config,
which usually provides better compatibility with the system in question.
But in some cases this doesn't provide the ideal configuration and users
might want to provide a more specific base kernel. Support this via new
KERNEL_HOST_CONFIG_TEMPLATE/KERNEL_GUEST_CONFIG_TEMPLATE environment
variables which can be set within stable-commits file.

Signed-off-by: Michael Roth <michael.roth@amd.com>

mdroth committed:
launch-qemu: use 'confidential-guest-support' instead of 'memory-encryption'

The older option is legacy and sometimes causes confusion amongst SNP
users trying to match up with command-lines they've seen elsewhere.

Signed-off-by: Michael Roth <michael.roth@amd.com>

mdroth committed:
launch-qemu: add -dry option for dry runs

Spitting out the command-line and taking no other action is useful in
some case.

Signed-off-by: Michael Roth <michael.roth@amd.com>

mdroth committed:
run make clean on edk2 BaseTools directory for every run

Pre-existing/pre-built BaseTools don't handle jumping around from one
edk2 version to the next very well, so just clean it up and regenerate
it every time. It's only a small portion of the overall package build
time anyway.

Signed-off-by: Michael Roth <michael.roth@amd.com>

tlendacky committed:
OVMF stable-commits update and documentation clarification

Update the OVMF repo to point to the upstream EDK2 repo and specify the
stable commit as the edk2-stable202502 commit.

Update the README to clarify that the guest install process should include
the serial console kernel parameter.

Signed-off-by: Tom Lendacky <thomas.lendacky@amd.com>

mdroth committed:
Fix git branch for edk2 builds

Running `./build --package` currently fails with:

  git checkout current/edk2-stable202502
  error: pathspec 'current/edk2-stable202502' did not match any file(s) known to git
  ERROR: git checkout current/edk2-stable202502

This is because stable-commits was updated to point to an edk2 stable
tag, but the scripts don't currently handle tags properly. Instead,
point to snp-latest branch for edk2, which is already updated to be a
mirror of the latest (at this time) edk2 stable tag.

Signed-off-by: Michael Roth <michael.roth@amd.com>

mdroth committed:
launch-qemu.sh: always set la57/phys-bits=52 CPU flags

For larger VMs, particularly ones with lots of GPUs with
system-addressable memory passed through that bring the total memory
close to ~1TB+, the default phys-bits setting quickly becomes
insufficient. Set a more appropriate default. While here, enable 5-level
paging support for better test coverage in that area (though for
phys-bits=52 it generally won't be used/needed).

Signed-off-by: Michael Roth <michael.roth@amd.com>

mdroth committed:
fix builds after a previous failure leaves kernel git tree unchecked out

The scripts assume cloning/reuse of a kernel git repo will result in a
Makefile being present, but a misnamed branch/etc. in a prior build can
put the working tree in a state where it will always fail trying to
distclean the source tree before checking it out. Fix this buying only
running distclean if there's a Makefile present to actually allow for it
to happen.

Signed-off-by: Michael Roth <michael.roth@amd.com>

arvindskumar99 committed:
versions: Update kernel to v6.16.1

Updating stable-commits to provide a way for
users to install the latest supported kernel
(v6.16.1) to match what is used in Kata and
CoCo.


Signed-off-by: Arvind Kumar <158101868+arvindskumar99@users.noreply.github.com>

ryansavino commented:
@arvindskumar99 this looks okay, but let's rebase the three branches in this repo with current AMDESE updates for this fork. It hasn't been updated since last November. Then let's put this on top and test/verify it before we merge it in.

mdroth committed:
common.sh: Exit with an error code if the build fails

Signed-off-by: Michael Roth <michael.roth@amd.com>

mdroth committed:
common.sh: Fix kernel config handling

Currently when specific kernel configs are specified in stable-commits,
a guest/host kernel config might clobber each other. Fix this by
handling each case appropriately.

Signed-off-by: Michael Roth <michael.roth@amd.com>

AdithyaKrishnan commented:
I'm hitting this error with your PR:
```
From https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux
 * tag                   v6.16.1    -> FETCH_HEAD
+ run_cmd git checkout v6.16.1
+ echo 'git checkout v6.16.1'
git checkout v6.16.1
+ eval 'git checkout v6.16.1'
++ git checkout v6.16.1
error: pathspec 'v6.16.1' did not match any file(s) known to git
+ echo 'ERROR: git checkout v6.16.1'
ERROR: git checkout v6.16.1
+ exit 1
```
